src/server/usage/README.md

```diff
@@ -0,0 +1,92 @@
+# Kibana Telemetry Service
+
+Telemetry allows Kibana features to have usage tracked in the wild. The general term "telemetry" refers to multiple things:
+
+1. Integrating with the telemetry service to express how to collect usage data (Collecting).
+2. Sending a payload of usage data up to Elastic's telemetry cluster, once per browser per day (Sending).
+3. Viewing usage data in the Kibana instance of the telemetry cluster (Viewing).
+
+You, the feature or plugin developer, mainly need to worry about the first meaning: collecting. To integrate with the telemetry services for usage collection of your feature, there are 2 steps:
+
+1. Create a usage collector using a factory function
+2. Register the usage collector with the Telemetry service
+
+NOTE: To a lesser extent, there's also a need to update the telemetry payload of Kibana stats and telemetry cluster field mappings to include your fields. This part is typically handled not by you, the developer, but different maintainers of the telemetry cluster. Usually, this step just means file an issue in the telemetry repo and have someone approve your data model.
+
+## Creating a Usage Collector
+
+A usage collector object is an instance of a class called `UsageCollector`. A factory function on `server.usage.collectorSet` object allows you to create an instance of this class. All you need to provide is a `type` for organizing your fields, and a `fetch` method for returning your usage data.
+
+Example:
+
+```js
+const myCollector = server.usage.collectorSet.makeUsageCollector({
+  type: MY_USAGE_TYPE,
+  fetch: async callCluster => {
+
+    // query ES and get some data
+    // summarize the data into a model
+    // return the modeled object that includes whatever you want to track
+
+    return {
+      my_objects: {
+        total: SOME_NUMBER
+      }
+    };
+  },
+});
+```
+
+Some background: The `callCluster` that gets passed to the `fetch` method is created in a way that's a bit tricky, to support multiple contexts the `fetch` method could be called. Your `fetch` method could get called as a result of an HTTP API request: in this case, the `callCluster` function wraps `callWithRequest`, and the request headers are expected to have read privilege on the entire `.kibana` index. The use case for this is stats pulled from a Kibana Metricbeat module, where the Beat calls Kibana's stats API in Kibana to invoke collection.
```

Comment:
This could be problematic with Kibana's new authorization model. If we are making this request on behalf of the end-user (i.e., `callWithRequest`), then we can't guarantee that the user will have _any_ access to the `.kibana` index. Is this use case only for the Metricbeat module, or is there potential for Kibana end users to end up in this scenario too?

Reply:
Currently, the usage data is made available in this API only for the Metricbeat module. The plan is to support an architecture where Monitoring collection is externalized and stats are shared over an HTTP API instead of collected internally. In this case, Metricbeat will have to be configured with a system role to call the API. The role to use might be built-in, but it will have to be documented.

cc @ycombinator